hypothesis-python/src/hypothesis/internal/conjecture/shrinker.py

```diff
@@ -1267,8 +1263,13 @@ def boost(k):
             if k > m:
                 return False
 
-            node_value = m - k
-            next_node_value = n + k
+            try:
+                node_value = m - k
+                next_node_value = n + k
+            except OverflowError:  # pragma: no cover
+                # if n or m is a float and k is over sys.float_info.max, coercing
+                # k to a float will overflow.
+                return False
```

Comment:
How well does this handle loss of precision?  For integer `k` I guess very small floats just round to the nearest resulting integer which seems fine, but for large `m` we might have `m - k == m and n + k > n` which makes this an antishrink pass in such cases!

(adding some type annotations for m, n, k might help clarify for future readers)

Reply:
good point, this will antishrink as you point out. Let's change this pass to only consider floats below MAX_PRECISE_INTEGER which should sidestep most of these issues

Reply:
Maybe worth noting that antishrinking shouldn't cause any problems, right? The shrink order means that we always only shrink, and cases where this ends up increasing shrink order should just get silently discarded. As long as those cases are relatively rare this is probably fine.

Reply:
Yeah, I think this is just a smallish performance issue, but worth the easy fix.